jetty-security/src/main/java/org/eclipse/jetty/security/authentication/ClientCertAuthenticator.java

```diff
@@ -113,12 +74,7 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
 
                 if (_validateCerts)
                 {
-                    KeyStore trustStore = getKeyStore(
-                        _trustStorePath, _trustStoreType, _trustStoreProvider,
-                        _trustStorePassword == null ? null : _trustStorePassword.toString());
-                    Collection<? extends CRL> crls = loadCRL(_crlPath);
-                    CertificateValidator validator = new CertificateValidator(trustStore, crls);
-                    validator.validate(certs);
+                    sslContextFactory.validateCerts(certs);
```

Comment:
If the no-args constructor is used this will always result in NPE from here?

Reply:
good catch. But I'm not sure what we return here as ssl is not configured and webapp requiring client-cert has been deployed. 400?

Reply:
If there is no trust store, then we just trust the certificate and perhaps verify is signature as a credential.
Question is, do we support a trust store being set either by a discovered sslContextFactory OR via a call to setTrustStore ?

I'm thinking we should leave and deprecated the existing class and just create a new one that we use if we have one and only one discovered sslContextFactory